lib/schema/src/validate.rs

```diff
@@ -208,6 +283,82 @@ impl ValidatedRegisterBlock {
             .collect();
     }
 
+    pub fn extract_array(&mut self, prefix: &str) -> Result<(), ValidationError> {
```

Comment:
Should we also be checking that `prefix` is reasonable, e.g., not an empty string, doesn't end with a number.

Reply:
I don't think it's necessary; the other checks will likely fail if there's not a contiguous set of registers.

An empty string might even be useful if the register block just has a bunch of registers with names like "0", "1", "2", "3".